### DISCUSSION TITLE
Multiple sources

### DISCUSSION THREAD

**Discussion by jellyedwards** [uber/react-map-gl#1710] [Q&A] [answered]
Hi, sorry if this is a dumb question but is it possible to add multiple sources to a map?

I tried:

```
    <Map
      width="100vw"
      height="100vh"
      mapStyle="https://api.maptiler.com/maps/basic/style.json"
      styleDiffing
    >
      <Source id="blue" type="geojson" data={geojson1}>
        <Layer {...blueStyle} />
      </Source>
      <Source id="red" type="geojson" data={geojson2}>
        <Layer {...redStyle} />
      </Source>
    </Map>
```

However they seem to overwrite so I only get one dataset shown.

[https://codesandbox.io/s/react-map-gl-multiple-sources-5jqgz](https://codesandbox.io/s/react-map-gl-multiple-sources-5jqgz)

Thanks!

**Comment by jellyedwards** [comment, marked as answer]
I had set the id of the two layers the same - d'oh!

**Comment by jashanbhullar** [comment]
@jellyedwards I am tryging do something similar but it's just overriding. 
https://codesandbox.io/p/devbox/react-map-gl-demo-qxrz7n

**Reply by jellyedwards** [reply]
very late to respond sorry! but it looks like your example works, did you get it fixed in the meantime?
